nupic/embodied/policies/curious_cnn_policy.py

```diff
@@ -0,0 +1,219 @@
+
+    Attributes
+    ----------
+    ac_pdtype : type
+        Description of attribute `ac_pdtype`.
+    pd : type
+        Description of attribute `pd`.
+    vpred : type
+        Description of attribute `vpred`.
+    features_model : torch.Sequential
+        Small conv net to extract features from observations.
+    pd_hidden : type
+        Hidden layer of the policy network of size hid_dim (2 layer, relu).
+    pd_head : type
+        Linear FC layer following pd_hidden with policy output.
+    vf_head : type
+        Linear FC layer following pd_hidden with value output (1).
+    param_list : type
+        List of parameters to be optimized.
+    flat_features : type
+        flattened feature vector.
+    ac : array
+        Current action.
+    ob : array
+        Current observation.
+
+    """
+
+    def __init__(
+        self,
+        ob_space,
+        ac_space,
+        ob_mean,
+        ob_std,
+        feat_dim,
+        hid_dim,
+        layernormalize,
+        nonlinear,
+        device,
+        scope="policy",
+    ):
+        if layernormalize:
+            print(
+                """Warning: policy is operating on top of layer-normed features.
+                It might slow down the training."""
+            )
+        self.layernormalize = layernormalize
+        self.nonlinear = nonlinear
+        self.ob_mean = ob_mean
+        self.ob_std = ob_std
+        self.ob_space = ob_space
+        self.ac_space = ac_space
+        # Get the type of probabiliyt distribution to use dependng on the environments
+        # action space type.
+        self.ac_pdtype = make_pdtype(ac_space)
+
+        self.pd = self.vpred = None
+        self.hid_dim = hid_dim
+        self.feat_dim = feat_dim
+        self.scope = scope
+        self.device = device
+        pdparamsize = self.ac_pdtype.param_shape()[0]
+
+        # Initialize the feature model as a small conv net (3 conv layer + 1 linear fc)
+        self.features_model = small_convnet(
+            self.ob_space,
+            nonlinear=self.nonlinear,
+            feat_dim=self.feat_dim,
+            last_nonlinear=None,
+            layernormalize=self.layernormalize,
+            batchnorm=False,
+            device=self.device,
+        ).to(self.device)
+
+        # Policy network following the feature extraction network (2 fc layers, relu)
+        self.pd_hidden = torch.nn.Sequential(
+            torch.nn.Linear(self.feat_dim, self.hid_dim),
+            torch.nn.ReLU(),
+            torch.nn.Linear(self.hid_dim, self.hid_dim),
+            torch.nn.ReLU(),
+        ).to(self.device)
+        # policy and value function head of the policy network.
+        self.pd_head = torch.nn.Linear(self.hid_dim, pdparamsize).to(self.device)
+        self.vf_head = torch.nn.Linear(self.hid_dim, 1).to(self.device)
+
+        # Define parameters to be optimized
+        self.param_list = [
+            dict(params=self.features_model.parameters()),
```

Comment:
This format is only required if you are specifying individual arguments (e.g. learning rate, momentum) for separate set of parameters. 

If not, you can just concatenate a list of Variables instead. So just make param_list a list, then you can call extend: `self.param_list.extend(new_param_list)`. Or during initialization use the spread operator:
```
self.param_list = [*self.features_model.parameters(), *self.pd_hidden.parameters(), ...]
```